Index: libgcrypt-1.4.0/mpi/config.links
===================================================================
--- libgcrypt-1.4.0.orig/mpi/config.links
+++ libgcrypt-1.4.0/mpi/config.links
@@ -237,11 +237,11 @@ case "${host}" in
        ;;
 
     ppc620-*-*	    | \
-    powerpc64*-*-*)
+    powerpc64-*-*)
 	mpi_sflags="-Wa,-mppc"
 	path="powerpc64"
 	;;
-    powerpc*-*-linux*)
+    powerpc-*-linux*)
 	echo '/* configured for powerpc/ELF */' >>./mpi/asm-syntax.h
 	echo '#define ELF_SYNTAX'               >>./mpi/asm-syntax.h
 	cat   $srcdir/mpi/powerpc32/syntax.h	>>./mpi/asm-syntax.h
@@ -276,7 +276,7 @@ case "${host}" in
 	mpi_sflags="-Wa,-mppc"
 	path="powerpc32"
 	;;
-    powerpc*-*-*)
+    powerpc-*-*)
 	mpi_sflags="-Wa,-mppc"
 	path="powerpc32"
 	;;
