diff -ur libgcrypt-1.5.0/cipher/cipher.c libgcrypt-1.5.0f/cipher/cipher.c
--- libgcrypt-1.5.0/cipher/cipher.c	2011-04-19 16:06:57.000000000 +0200
+++ libgcrypt-1.5.0f/cipher/cipher.c	2011-12-25 00:45:18.391287616 +0100
@@ -60,6 +60,12 @@
     { &_gcry_cipher_spec_blowfish,
       &dummy_extra_spec,                  GCRY_CIPHER_BLOWFISH },
 #endif
+#if USE_IDEA
+    { &_gcry_cipher_spec_idea,
+      &dummy_extra_spec,                  GCRY_CIPHER_IDEA },
+#else
+# warn IDEA not supported
+#endif
 #if USE_DES
     { &_gcry_cipher_spec_des,
       &dummy_extra_spec,                  GCRY_CIPHER_DES },
diff -ur libgcrypt-1.5.0/cipher/Makefile.am libgcrypt-1.5.0f/cipher/Makefile.am
--- libgcrypt-1.5.0/cipher/Makefile.am	2011-03-11 09:47:39.000000000 +0100
+++ libgcrypt-1.5.0f/cipher/Makefile.am	2011-12-25 00:39:27.051857406 +0100
@@ -51,6 +51,7 @@
 dsa.c \
 elgamal.c \
 ecc.c \
+idea.c \
 md4.c \
 md5.c \
 rijndael.c rijndael-tables.h \
diff -ur libgcrypt-1.5.0/configure.ac libgcrypt-1.5.0f/configure.ac
--- libgcrypt-1.5.0/configure.ac	2011-06-29 10:53:22.000000000 +0200
+++ libgcrypt-1.5.0f/configure.ac	2011-12-25 00:39:27.050857305 +0100
@@ -155,7 +155,7 @@
 LIBGCRYPT_CONFIG_HOST="$host"
 
 # Definitions for symmetric ciphers.
-available_ciphers="arcfour blowfish cast5 des aes twofish serpent rfc2268 seed"
+available_ciphers="arcfour blowfish cast5 des aes twofish serpent rfc2268 seed idea"
 available_ciphers="$available_ciphers camellia"
 enabled_ciphers=""
 
@@ -1005,6 +1005,12 @@
    AC_DEFINE(USE_CAST5, 1, [Defined if this module should be included])
 fi
 
+LIST_MEMBER(idea, $enabled_ciphers)
+if test "$found" = "1" ; then
+   GCRYPT_CIPHERS="$GCRYPT_CIPHERS idea.lo"
+   AC_DEFINE(USE_IDEA, 1, [Defined if this module should be included])
+fi
+
 LIST_MEMBER(des, $enabled_ciphers)
 if test "$found" = "1" ; then
    GCRYPT_CIPHERS="$GCRYPT_CIPHERS des.lo"
diff -ur libgcrypt-1.5.0/src/cipher.h libgcrypt-1.5.0f/src/cipher.h
--- libgcrypt-1.5.0/src/cipher.h	2011-06-10 10:53:41.000000000 +0200
+++ libgcrypt-1.5.0f/src/cipher.h	2011-12-25 00:40:25.323733275 +0100
@@ -118,6 +118,7 @@
 
 /* Declarations for the cipher specifications.  */
 extern gcry_cipher_spec_t _gcry_cipher_spec_blowfish;
+extern gcry_cipher_spec_t _gcry_cipher_spec_idea;
 extern gcry_cipher_spec_t _gcry_cipher_spec_des;
 extern gcry_cipher_spec_t _gcry_cipher_spec_tripledes;
 extern gcry_cipher_spec_t _gcry_cipher_spec_arcfour;
